xtask/src/main.rs

```diff
@@ -0,0 +1,605 @@
+                    None,
+                    &[],
+                )
+            },
+            "cross",
+            errors,
+        );
+    }
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv6m-none-eabi",
+                    "--workspace",
+                    "--exclude",
+                    "defmt-itm",
+                    "--exclude",
+                    "firmware",
+                ],
+                Some("firmware"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv7em-none-eabi",
+                    "--workspace",
+                ],
+                Some("firmware"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv6m-none-eabi",
+                    "--features",
+                    "print-defmt",
+                ],
+                Some("firmware/panic-probe"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command::<&str>(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv6m-none-eabi",
+                    "--features",
+                    "print-rtt",
+                ],
+                Some("firmware/panic-probe"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    )
+}
+
+fn test_snapshot(errors: &mut Vec<String>) {
+    println!("🧪 qemu/snapshot");
+    let mut tests = vec![
+        "log",
+        "timestamp",
+        "panic",
+        "assert",
+        "assert-eq",
+        "assert-ne",
+        "unwrap",
+        "defmt-test",
+        "hints",
+    ];
+
+    if rustc_is_nightly() {
+        tests.push("alloc");
+    }
+
+    let mut features_map = HashMap::new();
+    features_map.insert("alloc", "alloc");
+    let no_features = "";
+
+    for test in &tests {
+        let features = features_map.get(test).unwrap_or(&no_features);
+
+        do_test(
+            || test_single_snapshot(test, features, false),
+            "qemu/snapshot",
+            errors,
+        );
+        do_test(
+            || test_single_snapshot(test, features, true),
+            "qemu/snapshot",
+            errors,
+        );
+    }
+}
+
+fn main() -> Result<(), Vec<String>> {
+    let opt: Options = Options::from_args();
+    let mut all_errors: Vec<String> = vec![];
```

Comment:
Since all the code has access to `all_errors` anyways: What do you think about making it a global?
```rust
static mut all_errors: Vec<String> = vec![];
```

Reply:
it *does* make the code a lot shorter, but also requires `unsafe {}` :D 
I think it's one of those rare cases where global mutable state could be OK, so I've gone ahead and wrapped it in a `once_cell::sync::Lazy`

Reply:
Even though I brought it up, I am actually very unsure about this. Globals always feel like a bad design, but I'd agree that it seems to make sense in this case 🤔